### PR TITLE
🚧 Search and replace all equality checks with pattern matching

### LINKED ISSUE #101: 🚧 Search and replace all equality checks with pattern matching
### I have done the items below . . .

- [X] I have updated the title by replacing the '**_<title_**>' section.

### Description

Search through code base for all `==` and `!=` checks with reference types and replace them with `is null` and `is not null`.

This is for a bit of a performance boost but most of all it is for avoiding incorrect checks that can occur if the `==` or `!=` operators are overloaded.

### Acceptance Criteria

**This issue is finished when:**
- [ ] Code documentation added if required
- [ ] Unit tests added
- [ ] All unit tests pass

### ToDo Items

- [ ] Draft pull request created and linked to this issue
- [X] Priority label added to issue (**_low priority_**, **_medium priority_**, or **_high priority_**)
- [X] Issue linked to the proper project
- [X] Issue linked to proper milestone
- [ ] Unit tests have been written and/or adjusted for code additions or changes
- [ ] All unit tests pass

### Issue Dependencies

_No response_

### Related Work

_No response_

### PR DESCRIPTION
**Pull request by ljdarj**
### **Description:**

To allow this pull request to be merged, please make sure that the following items of this pull request are complete.

This pull request closes #101

✅The pull request head branch must be a feature branch with the syntax 'feature/<issue-num>-sync-testing'.  
✅The pull request base branch must be a 'main' or 'preview' branch..  
✅The pull request head branch contains a valid issue number.  
✅The pull request title matches the linked issue title exactly.  
✅The pull request assignees match the assignees of the issue.  
✅The pull request labels match the labels of the issue.  
✅The pull request organizational projects match the organizational projects of the issue.  
✅The pull requests milestone matches the milestone of the issue.  

---

### **Additional Info:**
1. Pull requests are automatically synced with the associated issue upon creation by one of the KinsonDigital workflow bots. 
2. The associated issue is the issue number that is embedded in the pull request head branch.
3. The list above will be automatically updated as the pull request's various settings match or do not match the associated issue.
4. To manually sync the pull request to the issue, create a comment with the `[run-sync]` command.
	> **Note** You must be an admin member of the organization to use this command.

---

- [x] Sync with the issue.  Use this to enable or disable this pull request from syncing with its associated issue.

### CODE REVIEW

**Comment by CalvinWilkinson** [comment]
@ljdarj Quick heads up.  The name of your branch is incorrect.  The branch is using underscores and they should by hyphens.

The branch should be named this:
```diff
- ljdarj:feature/101-equality_checks_to_pattern_matches
+ ljdarj:feature/101-equality-checks-to-pattern-matches
```

It is OK this time.  It did not fail the status checks.  Since the work has pretty much been finished, it would be too much work to have you rename/redo the branch. 😀

Just from now on, please make sure that the name of the branch is correct.  Some of the CICD infrastructure depends on it.

Here are some resources about contributing to KinsonDigital projects:
- [Contribution Guide](https://github.com/KinsonDigital/.github/blob/main/docs/CONTRIBUTING.md#contribution-guide)
- [Branching](https://github.com/KinsonDigital/.github/blob/main/docs/BranchingModel.md#branching) 

Are you finished with this issue by chance?  If you need more time, that is fine.  Just curious so I know which release to put this into.  😉

Cheers!!

**Comment by ljdarj** [comment]
I'm finished with it, it was rather short. I'll keep in mind your remarks, I misunderstood the point in the documentation as to the branches' naming convention. And sorry for not answering earlier, I was rather busy all week ^^;

**Comment by CalvinWilkinson** [comment]
> I'm finished with it, it was rather short. I'll keep in mind your remarks, I misunderstood the point in the documentation as to the branches' naming convention. And sorry for not answering earlier, I was rather busy all week ^^;

Thanks for the reply and your contribution!!  Not a problem.  We all have personal lives and other responsibilities more important than contributing to my little projects.  🤣

I will take a look at it.

If you want to contribute more, you are always welcome to!!

One thing you could do if you don't mind, is when you create a pull request, create the PR as a draft PR that way I know it is not ready yet.  Once you are finished, you can set the PR as ready for review which takes it out of draft mode. 

This is all built into GitHub and is there to let people know if something is still being worked on or is ready for review and is complete.

Thanks!!

File: Velaptor/AppStats.cs [hunk 1]
```diff
@@ -156,7 +156,7 @@ internal static void RemoveLoadedTexture(uint textureId)
     /// <param name="fontInfo">The loaded font info.</param>
     internal static void RecordLoadedFont(string fontInfo)
     {
-        if (string.IsNullOrEmpty(fontInfo) is not false || !fontInfo.Contains("|size:"))
```
Comment (KinsonDigitalAdmin) [inline Velaptor/AppStats.cs:159]: @ljdarj Good catch on this one!!

Funny how things like this make it into code.  😉

**Review by KinsonDigitalAdmin — APPROVED** [review]

**Pull request merged**